### PR TITLE
Added Level as prefix for loggers.

### PR DESCRIPTION
**Pull request by harshjain2**
Added Level as prefix for loggers.

### CODE REVIEW

File: test/Microsoft.TestPlatform.CoreUtilities.UnitTests/Output/OutputExtensionsTests.cs [hunk 1]
```diff
@@ -46,7 +48,7 @@ public void CleanUp()
         public void OutputErrorForSimpleMessageShouldOutputTheMessageString()
         {
             this.mockOutput.Object.Error("HelloError", null);
-            this.mockOutput.Verify(o => o.WriteLine("HelloError", OutputLevel.Error), Times.Once());
+            this.mockOutput.Verify(o => o.WriteLine("Error: HelloError", OutputLevel.Error), Times.Once());
```
Comment (smadala) [inline test/Microsoft.TestPlatform.CoreUtilities.UnitTests/Output/OutputExtensionsTests.cs]: Can we use  `Resources.CommandLineError` here?

**Comment by harshjain2** [comment]
@dotnet-bot test this.

**Comment by harshjain2** [comment]
@dotnet-bot test this please.

File: src/Microsoft.TestPlatform.CoreUtilities/Output/OutputExtensions.cs [hunk 2]
```diff
@@ -15,6 +15,13 @@ namespace Microsoft.VisualStudio.TestPlatform.Utilities
     public static class OutputExtensions
     {
         /// <summary>
+        /// Bool to decide whether Verbose level should be added as prefix or not in log messages.
+        /// </summary>
+        internal static bool AppendPrefix;
```
Comment (codito) [inline src/Microsoft.TestPlatform.CoreUtilities/Output/OutputExtensions.cs]: I am not sure if an Extension methods unit should contain state or configuration.

File: src/vstest.console/Internal/ConsoleLogger.cs [hunk 3]
```diff
@@ -150,6 +155,12 @@ public void Initialize(TestLoggerEvents events, Dictionary<string, string> param
                 this.verbosityLevel = verbosityLevel;
             }
 
+            var prefixExists = parameters.TryGetValue(ConsoleLogger.PrefixParam, out string prefix);
+            if (prefixExists)
+            {
+                bool.TryParse(prefix, out OutputExtensions.AppendPrefix);
```
Comment (codito) [inline src/vstest.console/Internal/ConsoleLogger.cs]: If this returns false, we don't need to log?

File: test/Microsoft.TestPlatform.CoreUtilities.UnitTests/Output/OutputExtensionsTests.cs [hunk 4]
```diff
@@ -48,7 +46,17 @@ public void CleanUp()
         public void OutputErrorForSimpleMessageShouldOutputTheMessageString()
         {
             this.mockOutput.Object.Error("HelloError", null);
+            this.mockOutput.Verify(o => o.WriteLine("HelloError", OutputLevel.Error), Times.Once());
+        }
+
+        [TestMethod]
+        public void OutputErrorForSimpleMessageShouldOutputTheMessageStringWithPrefixIfSet()
+        {
+            OutputExtensions.AppendPrefix = true;
```
Comment (codito) [inline test/Microsoft.TestPlatform.CoreUtilities.UnitTests/Output/OutputExtensionsTests.cs]: Do we need to validate other scenarios? Can it be a data driven test?

**Review by codito — CHANGES_REQUESTED** [review]

File: src/Microsoft.TestPlatform.CoreUtilities/Output/OutputExtensions.cs [hunk 5]
```diff
@@ -14,6 +14,11 @@ namespace Microsoft.VisualStudio.TestPlatform.Utilities
     /// </summary>
     public static class OutputExtensions
     {
+        /// <summary>
+        /// Bool to decide whether Verbose level should be added as prefix or not in log messages.
+        /// </summary>
+        internal static bool AppendPrefix;
```
Comment (codito) [inline src/Microsoft.TestPlatform.CoreUtilities/Output/OutputExtensions.cs]: Not sure if an extension methods class should have state.

**Comment by smadala** [comment]
Having prefix=true always makes finding warning and error messages easy, especially output is redirect to files in CI/CD scenarios. Currently user can identify the level of message by color of message in console which we loss once it redirect to file.

Update help for `vstest.console.exe` and `dotnet test`, update doc in vstest-docs.

**Review by Faizan2304 — APPROVED** [review]

**Review by smadala — APPROVED** [review]
Add doc in vstest-docs.

**Pull request merged**